### PR TITLE
feat: Added the ability to use the TEACH-ME mode

### LINKED ISSUE #211: Introduce teach-me mode
near-cli-rs can be an awesome tool to educate NEAR developers about how things actually work in NEAR.

For example, if would be great to provide more details about RPC calls that are executed and their parameters when `tokens view-ft-balance` is executed, which queries RPC for the token metadata and then the balance for the specific user. This would help answering questions like [this](https://stackoverflow.com/questions/76400166/how-can-i-query-fts-with-a-specified-account-name-using-rpc-call-on-near-protoco) much easier, I would just ask them to run near CLI with `--teach-me` flag.

Here is a rough illustration of the process when `near --teach-me` flag is set:

```
$ near --teach-me
> What are you up to? (select one of the options with the up-down arrows on your keyboard and press Enter) tokens      - Manage token assets such as NEAR, FT, NFT
> What is your account ID? frol.near
> Select actions with tokens: view-ft-balance   - View the balance of FT tokens
> What is the ft-contract account ID? wrap.near
> What is the name of the network? mainnet
> Сhoose block for view: now               - View properties in the final block
TEACH-ME: In order to display the FT symbol and format the decimal amount of owned tokens, we need to make a view-function call `ft_metadata` to the <wrap.near> contract, and we can do that with the query request to NEAR RPC node (https://docs.near.org/api/rpc/contracts#call-a-contract-function):
HTTP POST https://rpc.mainnet.near.org/
JSON-BODY: {
  "jsonrpc": "2.0",
  "id": "dontcare",
  "method": "query",
  "params": {
    "request_type": "call_function",
    "finality": "final",
    "account_id": "wrap.near",
    "method_name": "ft_metadata",
    "args_base64": "e30="
  }
}
RESPONSE: {
    "id": "dontcare",
    "jsonrpc": "2.0",
    "result": {
        "block_hash": "E4oQ8YhUcMSve7kRMvNeoM3ZSjJopJc1tPLztg3BWkmg",
        "block_height": 93496936,
        "logs": [],
        "result": [123,34,115,112,101,99,34,58,34,102,116,45,49,46,48,46,48,34,44,34,110,97,109,101,34,58,34,87,114,97,112,112,101,100,32,78,69,65,82,32,102,117,110,103,105,98,108,101,32,116,111,107,101,110,34,44,34,115,121,109,98,111,108,34,58,34,119,78,69,65,82,34,44,34,105,99,111,110,34,58,110,117,108,108,44,34,114,101,102,101,114,101,110,99,101,34,58,110,117,108,108,44,34,114,101,102,101,114,101,110,99,101,95,104,97,115,104,34,58,110,117,108,108,44,34,100,101,99,105,109,97,108,115,34,58,50,52,125]
    }
}
TEACH-ME: The `args_base64` (`e30=`) is a base64-encoded serialized arguments to `ft_metadata` function (`{}`), and the output `result` is an array of bytes (`{"spec":"ft-1.0.0","name":"Wrapped NEAR fungible token","symbol":"wNEAR","icon":null,"reference":null,"reference_hash":null,"decimals":24}`).

TEACH-ME: In order to get the FT balance for the specific account, we need to make a view-function call `ft_balance_of` to the <wrap.near> contract, and we can do that with the query request to NEAR RPC node (https://docs.near.org/api/rpc/contracts#call-a-contract-function):
HTTP POST https://rpc.mainnet.near.org/
JSON-BODY: {
  "jsonrpc": "2.0",
  "id": "dontcare",
  "method": "query",
  "params": {
    "request_type": "call_function",
    "finality": "final",
    "account_id": "wrap.near",
    "method_name": "ft_balance_of",
    "args_base64": "eyJhY2NvdW50X2lkIjogImZyb2wubmVhciJ9"
  }
}
RESPONSE: {
    "id": "dontcare",
    "jsonrpc": "2.0",
    "result": {
        "block_hash": "E4oQ8YhUcMSve7kRMvNeoM3ZSjJopJc1tPLztg3BWkmg",
        "block_height": 93496936,
        "logs": [],
        "result": [34, 57, 56, 55, 53, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 48, 34]
    }
}
TEACH-ME: The `args_base64` (`eyJhY2NvdW50X2lkIjogImZyb2wubmVhciJ9`) is a base64-encoded serialized arguments to `ft_balance_of` function (`{"account_id": "frol.near"}`), and the output `result` is an array of bytes (`"98750000000000000000000"`).

--------------
No logs
--------------

<frol.near> account has 0.09875 wNEAR  (FT-contract: wrap.near)
```

### Implementation hints

Follow `offline` flag implementation that is part of `GlobalContext` to add `--teach-me` flag throughout the project.

We can easily introduce TEACH-ME to the RPC helper functions:

https://github.com/near/near-cli-rs/blob/4f8576e1db857bdba45f9e3a57349253f99e07c2/src/common.rs#L1576-L1667

https://github.com/near/near-cli-rs/blob/4f8576e1db857bdba45f9e3a57349253f99e07c2/src/common.rs#L1688-L1767

That would already get us 80% through. Let's start with this and iterate from here.

### PR DESCRIPTION
**Pull request by FroVolod**
Resolves #211

### CODE REVIEW

**Comment by race-of-sloths** [comment]
@FroVolod Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/FroVolod?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/FroVolod?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/FroVolod)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/FroVolod)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by frol — DISMISSED** [review]
@race-of-sloths score 8

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

File: docs/GUIDE.en.md
Comment (frol) [inline docs/GUIDE.en.md:20]:
```suggestion
_near CLI_ is a great tool for understanding NEAR on the low level. For example, if you want to view more detailed information about the RPC calls being made and their parameters, simply run the CLI with the `--teach-me` flag:
```

File: docs/GUIDE.ru.md
Comment (frol) [inline docs/GUIDE.ru.md:18]:
```suggestion
_near CLI_ — отличный инструмент для глубокого понимания NEAR. Например, если вы хотите просмотреть более подробную информацию о выполняемых вызовах RPC и их параметрах, просто запустите CLI с флагом `--teach-me`:
```

**Review by frol — APPROVED** [review]

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 8!
You have received 105 (80 base + 15 weekly bonus + 10 monthly bonus) Sloth points for this contribution


Congratulations @FroVolod! Your PR was highly scored and you completed another monthly streak! To keep your monthly streak make another pull request next month and get 8+ score for it

**Pull request merged**
